packages/web3-utils/src/promise_helpers.ts

```diff
@@ -73,47 +73,56 @@ export async function waitWithTimeout<T>(
 	}
 	return result;
 }
+
+
 /**
  * Repeatedly calls an async function with a given interval until the result of the function is defined (not undefined or null),
- * or until a timeout is reached.
+ * or until a timeout is reached. It returns promise and intervalId.
  * @param func - The function to call.
  * @param interval - The interval in milliseconds.
  */
-export async function pollTillDefined<T>(
+export function pollTillDefinedAndReturnIntervalId<T>(
 	func: AsyncFunction<T>,
 	interval: number,
-): Promise<Exclude<T, undefined>> {
-	const awaitableRes = waitWithTimeout(func, interval);
+): [Promise<Exclude<T, undefined>>, Timer] {
 
 	let intervalId: Timer | undefined;
 	const polledRes = new Promise<Exclude<T, undefined>>((resolve, reject) => {
 		intervalId = setInterval(() => {
-			(async () => {
-				try {
-					const res = await waitWithTimeout(func, interval);
-
-					if (!isNullish(res)) {
-						clearInterval(intervalId);
-						resolve(res as unknown as Exclude<T, undefined>);
-					}
-				} catch (error) {
-					clearInterval(intervalId);
-					reject(error);
-				}
+			( () => {
+					
+					func()
+						.then( res =>{
+						if (!isNullish(res)) {
+							clearInterval(intervalId);
+							resolve(res as unknown as Exclude<T, undefined>);
+						}})
+						.catch(error=> {
+							clearInterval(intervalId);
+							reject(error);
+					});
+					
 			})() as unknown;
 		}, interval);
+
 	});
 
-	// If the first call to awaitableRes succeeded, return the result
-	const res = await awaitableRes;
-	if (!isNullish(res)) {
-		if (intervalId) {
-			clearInterval(intervalId);
-		}
-		return res as unknown as Exclude<T, undefined>;
-	}
+	return [polledRes, intervalId!];
+}
 
-	return polledRes;
+/**
+ * Repeatedly calls an async function with a given interval until the result of the function is defined (not undefined or null),
+ * or until a timeout is reached.
+ * pollTillDefinedAndReturnIntervalId() function should be used instead of pollTillDefined if you need IntervalId in result.
+ * This function will be deprecated in next major release so use pollTillDefinedAndReturnIntervalId(). 
+ * @param func - The function to call.
+ * @param interval - The interval in milliseconds.
+ */
+export async function pollTillDefined<T>(
```

Comment:
keeping `pollTillDefined` so we dnt break API as its exported function from utils.